### DISCUSSION TITLE
Using tectonic with TeXstudio

### DISCUSSION THREAD

**Discussion by zoof** [tectonic-typesetting/tectonic#1067] [Q&A]
Is this possible?  If so, how?  I did find [this](https://tex.stackexchange.com/questions/659497/tectonic-under-text-editors) but I think TeXstudio is different enough that it's not working for me.

**Comment by fridtjof** [comment]
<img width="911" alt="A screenshot of TeXStudio settings. The 'Build' section is currently selected. A user command called 'user0' is set up to run '/usr/bin/tectonic %.tex --keep-logs'. The default compiler is changed to 'txs:///user0', which will run the aforementioned user command to compile documents." src="https://github.com/tectonic-typesetting/tectonic/assets/2780577/f055797d-c362-4195-9a00-a29f8a306393">

I set it up like this (you might need to adjust the tectonic path depending on where it's installed)

**Reply by goyalyashpal** [reply]
but this would work only for single files right?

but the tectonic build command works on a directory i.e. the root `./` of the project, whereas, in txs, you'd likely be working on files in `./src/*.tex`

you'd have to make sure that tectonic build is invoked on the root itself....

**Reply by fridtjof** [reply]
Yeah... I just came back to this, tried it again and have no clue what I was thinking :/ @zoof can you somehow "unanswer" this post?

**Reply by korikhin** [reply]
Here is a quick guide in case someone still has issues:

1. Navigate to the `Build` tab in the settings.
2. Add a user command. E.g. `tectonic:Tectonic`

```sh
"/opt/homebrew/bin/tectonic" \
  -X compile %.tex \
  --only-cached \
  --keep-intermediates \
  --keep-logs \
  --synctex \
  --outdir build
```

- Notice that the full path of the executable is used. Adjust parameters to your needs.
- Specify a build directory to store build artifacts separately. (Optional)

3. Set the newly created command as the `Default Compiler` command.

- Done. 👍

4. Open `Advanced Options`.
5. Set the `build` directory as the search path for PDF and log files. (Optional)

![guide](https://github.com/user-attachments/assets/acafdcef-c2e2-45c3-b966-c634f2b40a8d)

**Reply by korikhin** [reply]
And for [Zed](https://github.com/zed-industries/zed) users:

1. Install [TexLab](https://github.com/latex-lsp/texlab).
2. Go to `Settings`.
3. Add the following configuration

```json
{
  "lsp": {
    "texlab": {
      "settings": {
        "texlab": {
          "build": {
            "executable": "tectonic",
            "onSave": true,
            "args": [
              "-X",
              "compile",
              "%f",
              "--only-cached",
              "--keep-intermediates",
              "--keep-logs",
              "--synctex",
              "--outdir",
              "build"
            ]
          }
        }
      }
    }
  }
}
```

**Comment by zoof** [comment]
So for BibTeX, do I need to have a basic set of TeXlive binaries installed?

**Reply by fridtjof** [reply]
No idea, I've only tested this with a texlive distribution still installed. Does it complain for you?

**Reply by zoof** [reply]
Interestingly, in a VM that I spun up to try, tectonic itself will call bibtex as necessary but F8, requires a local bibtex binary.  On Arch, this requires `texlive-bin` and `texlive-basic` and the command line needs in addition `--keep-intermediates` to keep the .aux file. I've also added `-Z search-path=<path>` for my own custom style and bibliography style files.
